packages/experiments-realm/simple-search-card.gts

```diff
@@ -0,0 +1,135 @@
+import {
+  contains,
+  field,
+  Component,
+  CardDef,
+  StringField,
+} from 'https://cardstack.com/base/card-api';
+import { commandData } from 'https://cardstack.com/base/resources/command-data';
+import {
+  SearchCardsResult,
+  SearchCardsByTypeAndTitleInput,
+} from 'https://cardstack.com/base/commands/search-card-result';
+import { SearchCardsByTypeAndTitleCommand } from '@cardstack/boxel-host/commands/search-cards';
+
+export class SimpleSearchCard extends CardDef {
+  static displayName = 'Simple Search';
+  @field titleSearch = contains(StringField);
+
+  static isolated = class Isolated extends Component<typeof this> {
+    searchResource = commandData<
+      typeof SearchCardsByTypeAndTitleInput,
+      typeof SearchCardsResult
+    >(this, SearchCardsByTypeAndTitleCommand, () => {
+      if (!this.args.model.titleSearch) {
+        return undefined; // not executing command
+      }
+      return new SearchCardsByTypeAndTitleInput({
+        title: this.args.model.titleSearch,
+      });
+    });
+
+    get searchResults() {
+      const resource = this.searchResource?.current;
```

Comment:
We have a question mark after this.searchResource in a bunch of places here. Is it required? Can searchResource every be undefined?